gui/backend/send_metric_event.go

```diff
@@ -0,0 +1,42 @@
+package backend
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"net/http"
+)
+
+type sendMetricEventRequest struct {
+	EventName string `json:"name"`
+	// TODO DS Populate request struct.
+}
+
+type sendMetricEventResponse struct {
+	Success bool `json:"success"`
+}
+
+func (s *APIServer) sendMetricEvent(w http.ResponseWriter, r *http.Request) {
+	bodyBytes, e := ioutil.ReadAll(r.Body)
+	if e != nil {
+		s.writeErrorJson(w, fmt.Sprintf("error reading request input: %s", e))
+		return
+	}
+	log.Printf("sendMetricEvent requestJson: %s\n", string(bodyBytes))
+
+	var req sendMetricEventRequest
+	e = json.Unmarshal(bodyBytes, &req)
+	if e != nil {
+		s.writeErrorJson(w, fmt.Sprintf("error unmarshaling json: %s; bodyString = %s", e, string(bodyBytes)))
+		return
+	}
+
+	e = s.metricsTracker.SendGuiEvent(req.EventName)
```

Comment:
the function signature should also take in a `map[string]interface{}`. but if we don't need it in this PR then can expand the signature later

Reply:
Agreed - there's a TODO for this above `SendGuiEvent`'s definition in `metricsTracker.go`.

Reply:
Updated: This function no longer exists. Instead, `metricsTracker.SendEvent` already takes an arbitrary interface.